code/modules/atmospherics/machinery/components/tank.dm

```diff
@@ -151,28 +154,53 @@
 	refresh_window()
 
 ///////////////////////////////////////////////////////////////////
-// Pipenet stuff
+// Port stuff
+
+/**
+ * Enables/Disables a port direction in [./var/open_ports]. \
+ * Use this, then call [./proc/set_init_directions()] instead of setting initialize_directions directly \
+ * This system exists because having some initialize_directions set to null breaks rotation
+ */
+/obj/machinery/atmospherics/components/tank/proc/set_portdir_relative(relative_port_dir, enable)
+
+	if(isnull(enable)) //toggle it if we weren't given a state to set it to
```

Comment:
add a new proc that does this, do not make toggling implicit behavior off null it mixes metaphor